src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -951,14 +951,55 @@ private void buildDockerImage(File tempDockerfile, File userDockerfile) throws P
             debug("Docker build context: " + userDockerfile.getParent());
             buildCmd = "docker build -f " + tempDockerfile + " -t " + imageName + " " + userDockerfile.getParent();
             info(buildCmd);
-            String buildOutput = execDockerCmd(buildCmd, dockerBuildTimeout);
-            debug("Docker build output: " + buildOutput);
+            long startTime = System.currentTimeMillis();
+            runCommandAndLog(getRunProcess(buildCmd), dockerBuildTimeout);
+            checkDockerIgnore(startTime, userDockerfile);
+            info("Completed building Docker image.");
+        } catch (IllegalThreadStateException  e) {
+            // the timeout was too short and the docker command has not yet completed.
+            debug("IllegalThreadStateException, message="+e.getMessage());
+            error("The docker build command did not complete within the timeout period: " + dockerBuildTimeout + " seconds. " +
+                "Use the dockerBuildTimeout option to specify a longer period or " +
+                "add files not needed in the container to the .dockerignore file.", e);
+            throw new PluginExecutionException("Could not build Docker image using Dockerfile: " + 
+                userDockerfile.getAbsolutePath() + ". Address the following docker build error and then start dev mode again: " +
+                "The docker build command did not complete within the timeout period: " + dockerBuildTimeout + " seconds", e);
```

Comment:
Is there a benefit of providing two different messages for `error` and `PluginExecutionException` (compared to making them the same message, or just combining them into the PluginExecutionException)?

Reply:
My thinking is that the first error message has all the details and when you review it you see it in context. The exception message is seen at the end so I made it more general. It includes the docker file that you might have seen in the context of the original message. I agree that it is quite wordy as it is. Perhaps in the case of a plugin exception we're going to halt execution so the distinction I just made is not so important.

Reply:
Combined the messages into the exception only.